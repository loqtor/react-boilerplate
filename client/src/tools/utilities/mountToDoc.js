/*
 * Utility to provide aXe with a fake DOM to allow for accessibility testing
 * on mounted components within our Jest tests
*/
import { mountWithIntl } from './enzymeIntlTestHelper';

let wrapper;

export default function mountToDoc(component) {
  const container = mountWithIntl(component);

  if (!wrapper) {
    wrapper = document.createElement('main');
    document.body.appendChild(wrapper);
  }

  wrapper.innerHTML = '';
  wrapper.appendChild(container.getDOMNode());

  return container;
}
