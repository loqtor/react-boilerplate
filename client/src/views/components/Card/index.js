import React from 'react';
import PropTypes from 'prop-types';

export const Card = ({ className, children }) => <div className={`Card ${className}`}>{children}</div>;

Card.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

Card.defaultProps = {
  className: '',
};
