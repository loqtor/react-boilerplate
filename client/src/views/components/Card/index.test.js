import React from 'react';
import { shallow } from 'enzyme';
import axe from 'axe-core';
import toJson from 'enzyme-to-json';

import mountToDoc from '../../../tools/utilities/mountToDoc';

import Card from './index';

const setup = (render) => {
  const component = render(<Card>Card text </Card>);

  return {
    actual: component,
  };
};

describe('index.test.jsx', () => {
  it('renders correctly', () => {
    const { actual } = setup(shallow, {});
    expect(toJson(actual)).toMatchSnapshot();
  });

  it('has no accessibility violations', (done) => {
    const { actual } = setup(mountToDoc, {});
    const componentNode = actual.getDOMNode();

    axe.run(componentNode, (err, { violations }) => {
      expect(err).toBe(null);
      expect(violations).toHaveLength(0);
      done();
    });
  });
});
