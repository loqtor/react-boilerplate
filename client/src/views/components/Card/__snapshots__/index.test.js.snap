// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.test.jsx renders correctly 1`] = `
<div
  className="Card "
>
  Card text 
</div>
`;
